cpp/examples/tutorial_examples/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+project(MyExample)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra")
+find_package(Arrow REQUIRED)
+get_filename_component(ARROW_CONFIG_PATH ${Arrow_CONFIG} DIRECTORY)
+find_package(Parquet REQUIRED HINTS ${ARROW_CONFIG_PATH})
+find_package(ArrowDataset REQUIRED HINTS ${ARROW_CONFIG_PATH})
+
+add_executable(arrow_example arrow_example.cc)
+target_link_libraries(arrow_example PRIVATE arrow_shared)
```

Comment:
With the new changes to our CMake build we are recommending to use `Arrow::arrow_shared` instead of `arrow_shared`, which is now only supported for backwards compatibility, see: https://github.com/apache/arrow/blob/master/docs/source/cpp/build_system.rst#available-variables-and-targets
Should we update the target link library names on the example, this will only be compatible for arrow >= 10.0.0?
Also `Parquet::parquet_shared` and `ArrowDataset::arrow_dataset_shared`

Reply:
I think I'll update it, but there should be a comment talking about proper practice for the old versions.

Reply:
I was able to get Arrow::arrow_shared to import, but not the other two. Not exactly sure what went wrong, the targets do seem to be there after i updated and built Arrow.

Reply:
Did it. Can't run it locally anymore, but it does work in the Docker image. Probably have to rebuild some things on my end.